synapse/lib/lmdbslab.py

```diff
@@ -363,8 +376,97 @@ def _growMapSize(self, size=None):
         self.lenv.set_mapsize(mapsize)
         self.mapsize = mapsize
 
+        self.resizeevent.set()
+
         return self.mapsize
 
+    def _memorylockloop(self):
+        '''
+        Separate thread loop that manages the prefaulting and locking of the memory backing the data file
```

Comment:
There is potentially a bunch of really useful debug/statistics information available in this function. It's worth keeping a dictionary struct updated with it (and any possible additional event indicating whether or not we've locked memory or not).  That data should in turn be exposed via the lmdblayer.stat() api.

Reply:
The only useful thing I think maybe is going to be prefault progress, and it is only going to be useful for at most 10 minutes at the start of a cortex. Perhaps the current amount locked and the max calculation would be useful

Reply:
it would also be useful during a resize event; the boolean idea of whether or not we are actively prefaulting or not; etc.